app/src/main/java/ch/epfl/sweng/SDP/Matchmaking/User.java

```diff
@@ -0,0 +1,181 @@
+package ch.epfl.sweng.SDP.Matchmaking;
+
+import android.annotation.TargetApi;
+import android.os.Build;
+
+import android.util.Log;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import static java.lang.Math.toIntExact;
+
+public class User implements Serializable {
+
+    private String name;
+    private String username;
+    private String id;
+
+    private int stars;
+    private int matchesWon;
+    private int matchesLost;
+    private int matchesPlayed;
+    private double averageRating;
+    private int trophies;
+
+    private ArrayList<String> friends;
+
+    private DatabaseReference database;
+
+    public User(String id) {
+        this.id = id;
+    }
+
+    public User() {
+        database = FirebaseDatabase.getInstance().getReference();
+        this.id = FirebaseAuth.getInstance().getCurrentUser().getUid();
+    }
+
+    /**
+     * Download an object from the database and set the variables of this object.
+     */
+    public void downloadUser() {
+
+        // Read from the database
+        database.child("users").child(id).addListenerForSingleValueEvent(new ValueEventListener() {
+
+
+            @TargetApi(Build.VERSION_CODES.N)
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                // This method is called once with the initial value and again
+                // whenever data at this location is updated.
+                HashMap<String, Object> map = (HashMap<String, Object>) dataSnapshot.getValue();
+                // Update room
+
+                name = (String) map.get("name");
+                name = (String) map.get("username");
+                id = (String) map.get("id");
+
+                stars = toIntExact( (Long) map.get("stars"));
+                trophies = toIntExact((Long) map.get("trophies"));
+                matchesWon = toIntExact( (Long) map.get("matchesWon"));
+                matchesLost = toIntExact((Long) map.get("matchesLost"));
+                matchesPlayed = toIntExact((Long) map.get("matchesPlayed"));
+                averageRating = toIntExact((Long) map.get("averageRating"));
```

Comment:
How are you going to calculate the average rating when a new rating is available? ((averageRating*matchesPlayed)+newRating)/(matchesPlayed+1)? Is a bit complicated.
Maybe save totalRating in database and display it like this: totalRanking/matchesPlayed